### ISSUE TITLE
No Java Runtime Environment

### ISSUE THREAD

**Issue by veganaize** [argouml-tigris-org/argouml#50]
ArgoUML 0.35.1
OS: Windows 8.1 (x64)

![image](https://user-images.githubusercontent.com/7102064/161583715-68b68b36-4079-426e-ba85-1702cda9e1c5.png)

Setup pops up an info dialog with: _**"No Java Runtime Environment could be found.  Setup will automatically download a suitable JRE from sun.com."**_  --even though I have a JRE installed.

```
C:\> echo %JAVA_HOME%
c:\progra~1\java\jdk1.8.0_202

C:\> where java
c:\Program Files\Java\jdk1.8.0_202\bin\java.exe
```
...I've also tried the `jre\` variations but the result was the same.

**Comment by linustolke** [comment]
I think this is a duplicate of #36. I will close this issue as a duplicate.

**Comment by veganaize** [comment]
> The setup.exe is very old and the scripts that made the build are no longer used. We should use a newer Maven-based builder for the Windows installation. There is a lot of work to set such a tool up. Please help if you can with this.

I totally would but [repo](https://gerrit.googlesource.com/git-repo/) doesn't support Windows 8.1 _(which is a bit ridiculous since even Microsoft still supports it!)_

**Comment by linustolke** [comment]
That is unfortunate. I assume you will move to Windows 10 or Windows 11 before January so you could perhaps get back to this then.

**Comment by veganaize** [comment]
I will absolutely _**not**_ be moving to Windows 10 or 11. :alien:

I'll be returning to GNU/Linux whenever my essential 3rd-party tools decide to abandon their compatibility with Windows 8.1... maybe a year or two from now?

I'm currently researching modern build tools & practices.  I feel like there may be better options available, which should work well for _all_ of the developers on the project.

_I will make sure my proposal includes only processes & tooling that that are greater than or equal to those currently being utilized by the project maintainers, and that the burden of adoption is minimized -- ie. I'll handle any heavy lifting._

**Comment by linustolke** [comment]
Aha, OK.

I don't really understand what work you are referring to or what proposal you are working with. If you expect me to, please be a little more specific otherwise I will assume that it doesn't concern me or the ArgoUML project.

**Comment by linustolke** [comment]
It sounds interesting though...

**Comment by veganaize** [comment]
Sorry.  I was implying that I might come back around and propose an alternative build process.

**Comment by linustolke** [comment]
Please let me know if you want me to go through the use cases or requirements.

**Comment by veganaiZe** [comment]
_Please do._

**Comment by linustolke** [comment]
I realize now that my statement is rather unclear. What I meant to say was that I can review the use cases and requirements that you have identified.

On the other hand, I could write something down also.

**Comment by veganaiZe** [comment]
Oh, I see...  :smile:

Maybe if you could briefly explain some reasoning behind the choices, for the current build tools?  _(repo, maven, etc.)_  If this has been explained elsewhere a link to that information would be appreciated.

Which affordances does the current tooling offer that should be carried forward, by any new tooling?  _Specific use cases & requirements are welcome._

**Comment by veganaiZe** [comment]
_Actually, scratch all of that!_

I can probably leave most/all of the current tools intact.  I need to continue looking into this, on my end.

Sorry to bother you with all of this.  If you don't hear from me for a while (ie. a month or more) then please feel free to ping me.

**Comment by linustolke** [comment]
It is no bother at all. I am as eager as you to improve the way we work.

Maven was suggested for use in the project in July 2005 when I complained about the ant build not integrating well with Eclipse. In December 2006 it was suggested again but that we should wait for Maven 2 to be released since it would be incompatible with Maven. I was a little reluctant since it involved moving around the code from an existing directory structure to a new directory structure but I think we got some basic things in place in 2008.

When moving to github I wanted an easy integration with Jenkins for the builds so I abandoned the old build scripts and fixed things in maven. I am still struggling with it and I am struggling with github. I think I am slowly getting to a better state.
